### DISCUSSION TITLE
How to tell netavark to use firewalld instead of iptables

### DISCUSSION THREAD

**Discussion by unknowndevQwQ** [containers/netavark#601] [Q&A] [answered]
In the readme.md page, I saw that netavark supports both firewalld and iptables, but when using it, I found that netavark prefers iptables to firewalld. I tried to find the answer in the netavark documentation, but this did not use

**Comment by Luap99** [comment, marked as answer]
you can overwrite it with the `NETAVARK_FW` env var
https://github.com/containers/netavark/blob/ce1e004710ebd48fd6542880161a8088c4528abb/src/firewall/mod.rs#L40

Ideally we would add this as field in containers.conf, see https://github.com/containers/common/issues/1338

**Reply by unknowndevQwQ** [reply]
> you can overwrite it with the `NETAVARK_FW` env var
> 
> https://github.com/containers/netavark/blob/ce1e004710ebd48fd6542880161a8088c4528abb/src/firewall/mod.rs#L40
> 
> Ideally we would add this as field in containers.conf, see [containers/common#1338](https://github.com/containers/common/issues/1338)

Ok, I got it. Also one question:Why does netavark call iptables and nftables firewalls instead of packet filters?

**Reply by unknowndevQwQ** [reply]
@Luap99 Can you see it?

**Reply by Luap99** [reply]
Firewall is common term most people understand. Packet filter may be correct for nftables and iptables but it would not match firewalld.

**Reply by unknowndevQwQ** [reply]
> Firewall is common term most people understand. Packet filter may be correct for nftables and iptables but it would not match firewalld.

If I want to implement a feature that sets the firewall via command parameters, should I name it firewall-backend or firewall-driver?

**Reply by Luap99** [reply]
Why do you want to set it via command line? We only support the use with podman and podman could just set the env so there is no need to add more code here.

**Reply by unknowndevQwQ** [reply]
> Why do you want to set it via command line? We only support the use with podman and podman could just set the env so there is no need to add more code here.

One of the reasons for adding the command line option is for possible other programs to work well with netavark in the future, and secondly, the debug information currently output by podman never outputs env when executing commands, which is potentially troublesome for troubleshooting problems.

**Reply by unknowndevQwQ** [reply]
@Luap99 I want to do this thing, is it okay?

**Reply by Luap99** [reply]
Yes

**Reply by unknowndevQwQ** [reply]
Why is the noun driver instead of backend when netavark calls firewalld/ipt/nft?
